include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -625,12 +625,12 @@ __parallel_radix_sort_iteration(_ExecutionPolicy&& __exec, ::std::size_t __segme
 
     // correct __block_size, __scan_wg_size, __reorder_sg_size after introspection of the kernels
 #if _ONEDPL_COMPILE_KERNEL
-    auto __count_kernel =
-        __internal::__kernel_compiler<_RadixCountKernel>::__compile_kernel(::std::forward<_ExecutionPolicy>(__exec));
-    auto __local_scan_kernel = __internal::__kernel_compiler<_RadixLocalScanKernel>::__compile_kernel(
-        ::std::forward<_ExecutionPolicy>(__exec));
-    auto __reorder_kernel =
-        __internal::__kernel_compiler<_RadixReorderKernel>::__compile_kernel(::std::forward<_ExecutionPolicy>(__exec));
+    auto __kernels =
+        __internal::__kernel_compiler<_RadixCountKernel, _RadixLocalScanKernel, _RadixReorderKernel>::__compile(
+            ::std::forward<_ExecutionPolicy>(__exec));
```

Comment:
we already discussed that forwarding ExecutionPolicy if you are using it below is dangerous.